### PR TITLE
Add note about pointer state after the call.

### LINKED ISSUE #23422: heap::reallocate doesn't specify state of old pointer on success

### PR DESCRIPTION
**Pull request by steveklabnik**
Fixes #23422

### CODE REVIEW

**Comment by alexcrichton** [comment]
Some reallocations can succeed in-place I believe, so in that sense this may not be 100% true because the same pointer could be returned. Perhaps something like:

> The memory at the passed-in pointer is undefined after the call if the allocation was relocated.

**Comment by steveklabnik** [comment]
@alexcrichton fixed, i just re-ordered the sentence

File: src/liballoc/heap.rs [hunk 1]
```diff
@@ -26,6 +26,10 @@ pub unsafe fn allocate(size: usize, align: usize) -> *mut u8 {
 ///
 /// On failure, return a null pointer and leave the original allocation intact.
 ///
+/// If the allocation was relocated, the memory at the passed-in pointer is undefined after the
+/// call.
+///
```
Comment (alexcrichton) [inline src/liballoc/heap.rs]: stray double newline, I'd also tend to preserve the surrounding style by wrapping to 80 chars as well. Other than that though r=me

Comment (steveklabnik) [inline src/liballoc/heap.rs]: Our vim files auto-wrap at 100, so doing this is a manual thing, which kinda sucks. I'll still do it here, this just keeps cropping up.

**Pull request merged**